Change imports to use 'fms' module

rem1776 opened the pull request:
Changes all fms imports (except old io) to use the 'fms' module instead.

needs [#729](https://github.com/NOAA-GFDL/FMS/pull/729) first

thomas-robinson commented on full/atm_land_ice_flux_exchange.F90:
```diff
@@ -113,6 +64,56 @@ module atm_land_ice_flux_exchange_mod
+  use fms, only: ISTOCK_SIDE, ISTOCK_TOP, ISTOCK_BOTTOM
+  use fms, only: stocks_file, stocks_report, stocks_report_init
+  use fms, only: Atm_stock, Ocn_stock, Lnd_stock, Ice_stock
+  use fms, only: CMOR_MISSING_VALUE, null_axis_id, clock_flag_default
```
I would prefer `use FMS`, but it really doesn't matter

bensonr replied:
If I'm interpreting @thomas-robinson correctly, I agree that our goal is a simple "use FMS" without restriction.

rem1776 replied:
@bensonr Do you mean using the whole fms namespace(without the onlys)? I think Tom might've just meant the capitalization.

bensonr replied:
@rem1776 - just like NetCDF, ESMF, and MPI, we should expose the namespace.  As for capitalization, that's is purely aesthetic in fortran.

rem1776 replied:
It's now updated with those changes. I needed to import some separately in atm_land_ice_flux_exchange.F90 but made note if it [here](https://github.com/rem1776/FMScoupler/blob/ef4ab01a053bb5b6b145f271b972554555c55830/full/atm_land_ice_flux_exchange.F90#L84-L93).

rem1776 commented:
@thomas-robinson This is ready for review, passed CI script with that other pr and some tests on gaea

bensonr reviewed (changes requested):
You can safely remove any the old CVS-style $Id$ and $Name$ and replace them as has been done in the FMS modules as well as the component models.  The standard has been:
 
modname="<file>_MOD"
#include <file_version.h>
call write_version_number (modname, version)

I don't know how we are going to handle the file_version.h in the future as it is embedded in different components to allow them to list the hash or tag of the release they are using.

rem1776 commented:
@bensonr I can compile with those changes if I add in the file_version.h from FMS, but then anything that compiles coupler would need an extra flag for the included file right? Wouldn't that cause issues with model runs unless the flag is added to the xmls?

Also write_version_number is from the old io so it's not included in the global FMS module, should a version be added to fms2_io?

bensonr commented:
@rem1776  definitely need a write_version_number available.  If not, you will break code throughout FMS and the various components.

rem1776 commented:
@bensonr Just posted [#735](https://github.com/NOAA-GFDL/FMS/pull/735) to add in write_version_number to fms_mod and libFMS

bensonr commented on full/coupler_main.F90:
```diff
@@ -313,47 +313,11 @@ program coupler_main
 
-  use fms_affinity_mod,        only: fms_affinity_init, fms_affinity_set
+  !< Can't get rid of this until fms_io is no longer used at all
+  use fms_io_mod,              only: fms_io_exit, write_version_number
```
Once write_version_number is moved to fms_mod and is publicized via 'use FMS', this will need to be modified.

bensonr commented on full/flux_exchange.F90:
```diff
@@ -516,34 +503,13 @@ module flux_exchange_mod
+  use FMS, tfreeze_fms => tfreeze, version_fms => version
+
+  !! fms_io not in global module(unsupported)
+  use fms_io_mod, only:  write_version_number
```
Once write_version_number is moved to fms_mod and is publicized via 'use FMS', this will need to be modified.

bensonr commented on shared/surface_flux.F90:
```diff
@@ -113,12 +113,9 @@
-use             mpp_mod, only: input_nml_file, FATAL, mpp_error
+use FMS, version_fms=>version, kappa_fms=>kappa, epsln_fms=>epsln
+
+use fms_io_mod, only: write_version_number, file_exist, open_namelist_file
```
Once write_version_number is moved to fms_mod and is publicized via 'use FMS', this will need to be modified.  I believe file_exist and open_namelist_file can be replaced with the fms2_io equivalents.

bensonr commented on simple/coupler_main.F90:
```diff
@@ -53,51 +53,16 @@ program coupler_main
+use FMS, version_fms=>version
 
+!--- FMS old io
+use fms_io_mod, only: write_version_number, fms_io_exit!< This can't be removed until fms_io is not used at all
```
Once write_version_number is moved to fms_mod and is publicized via 'use FMS', this will need to be modified.

bensonr commented on simple/coupler_main.F90:
```diff
@@ -53,51 +53,16 @@ program coupler_main
+                              flux_up_to_atmos!,     &
+                              !flux_exchange_end       ! may not be used?
+!--- FMS modules
+use FMS, version_fms=>version
```
Not sure we want to publicize version from FMS simply as 'version'.  It probably should be exported as version_fms from the beginning.

bensonr commented on simple/flux_exchange.F90:
```diff
@@ -21,38 +21,20 @@
+!! FMS
+use FMS, version_fms=>version
 
-use field_manager_mod,  only: MODEL_ATMOS
-use tracer_manager_mod, only: get_number_tracers, get_tracer_index, NO_TRACER
+!! FMS old io
+use fms_io_mod, only: write_version_number
```
same comments as before

bensonr commented on simple/ice_model.F90:
```diff
@@ -21,26 +21,16 @@
-                            mpp_get_global_domain, mpp_get_compute_domain
-use mpp_mod, only: mpp_min, mpp_max, input_nml_file
+!! FMS
+use FMS, version_fms=>version, sst_anom_fms=>sst_anom
+
+!! FMS old io
+use fms_io_mod, only: write_version_number
```
same comment

rem1776 commented:
@bensonr I removed the imports for write_version_number and version to go along with the FMS updates in [#735](https://github.com/NOAA-GFDL/FMS/pull/735/) and also replaced the fms_io calls

bensonr reviewed (commented):
I think you should only 'use' FMSconstants when needed, not everywhere by default.

rem1776 commented:
@bensonr I believe it is needed in all of these files, I only added it if there was a `use constants_mod` previously. Do you mean only the specific parameters?

bensonr commented:
>
> @bensonr <https://github.com/bensonr> I believe it is needed in all of
> these files, I only added it if there was a use constants_mod previously.
> Do you mean only the specific parameters?
>
I was commenting on blanket use.  If it was there previously, then yes it
should be there now.

rem1776 commented:
@bensonr This should be good now, I added in the original only lists and removed the aliases that were added in

rem1776 commented:
Needed one more line in libFMS.F90 to make the generic version defined within file_version.h private, just posted [#737](https://github.com/NOAA-GFDL/FMS/pull/737) with the change. Passed CI with that PR.
